frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs

```diff
@@ -0,0 +1,202 @@
+  :: forall key t m mConf
+  . ( MonadWidget t m
+    , HasUISigningModelCfg mConf key t
+    , HasCrypto key (Performable m)
+    )
+  => AppCfg key t m
+  -> ModalIde m key t
+  -> Event t ()
+  -> m (mConf, Event t ())
+uiAddVanityAccount _appCfg ideL _onCloseExternal = do
+  rec
+    onClose <- modalHeader $ dynText title
+    result <- workflow $ uiAddVanityAccountSettings ideL
+    let (title, (eNewAccount0, conf0)) = fmap splitDynPure $ splitDynPure result
+
+  conf <- flatten =<< tagOnPostBuild conf0
+  let eNewAccount = switch $ current eNewAccount0
+  pure ( conf & walletCfg_importAccount .~ eNewAccount
+       , leftmost [() <$ eNewAccount, onClose]
+       )
+
+tempkeyset :: Text
+tempkeyset = "temp-vanity-keyset"
+
+mkPubkeyPactData :: KeyPair key -> Object
+mkPubkeyPactData  = HM.singleton tempkeyset . Array . V.singleton . String . keyToText . _keyPair_publicKey
+
+mkPactCode :: Maybe AccountName -> Text
+mkPactCode (Just acc) = "(coin.create-account \"" <> unAccountName acc <> "\" (read-keyset \"" <> tempkeyset <> "\"))"
+mkPactCode _ = ""
+
+uiAddVanityAccountSettings
+  :: forall key t m mConf
+  . ( MonadWidget t m
+    , HasUISigningModelCfg mConf key t
+    , HasCrypto key (Performable m)
+    )
+  => ModalIde m key t
+  -> Workflow t m (Text, (Event t (Account key), mConf))
+uiAddVanityAccountSettings ideL = Workflow $ do
+  pb <- getPostBuild
+
+  let inputElem lbl wrapperCls = divClass wrapperCls $ flip mkLabeledClsInput lbl
+        $ \cls -> uiInputElement $ def & initialAttributes .~ "class" =: (renderClass cls)
+
+      w = _ide_wallet ideL
+      dNextKey = findNextKey w
+
+      validateAccountName = checkAccountNameValidity w
+
+      uiAccountNameInput = divClass "vanity-account-create__account-name" $ do
+        dEitherAccName <- (validateAccountName <*>) . value <$>
+          inputElem "Account Name" "vanity-account-create__account-name-input"
+
+        dAccNameDirty <- holdUniqDyn =<< holdDyn False (True <$ updated dEitherAccName)
+
+        divClass "vanity-account-create__account-name-error" $
+          dyn_ $ ffor2 dAccNameDirty dEitherAccName $ \d e -> if d then either text (const blank) e else blank
+
+        pure $ hush <$> dEitherAccName
+
+      uiAcc = Just $ ("Reference Data",) $ liftA2 (,) uiAccountNameInput
+        (value <$> inputElem "Notes" "vanity-account-create__notes")
+
+  eKeyPair <- performEvent $ cryptoGenKey <$> current dNextKey <@ pb
+  dKeyPair <- holdDyn Nothing (Just . (\(pr,pu) -> KeyPair pu (Just pr)) <$> eKeyPair)
+
+  rec
+    (curSelection, eNewAccount, _) <- buildDeployTabs Nothing controls
+
+    (conf, result, dAccount) <- elClass "div" "modal__main transaction_details" $ do
+      (cfg, cChainId, ttl, gasLimit, mAccountDetails) <- tabPane mempty curSelection DeploymentSettingsView_Cfg $
+        -- Is passing around 'Maybe x' everywhere really a good way of doing this ?
+        uiCfg Nothing ideL (userChainIdSelect ideL) Nothing (Just defaultTransactionGasLimit) uiAcc
```

Comment:
Should we allow progress past this point without selecting a chain? 
If not, we could do 
```haskell
dmdChainId <- maybeDyn mChainId
ffor dmdChainId $ \case
  Nothing -> blank
  Just cChainId-> everythingPastThisPoint
```

Reply:
For now you cannot proceed without selecting a chainId because if you don't then the signing/capabilities allocation won't let you pass and it tells you to go pick a chain.

I'm not against the change, but we'd need to consider all of the deployment processes and make the change in a more core location.